staging/src/k8s.io/apiserver/pkg/admission/metrics/metrics.go

```diff
@@ -46,7 +46,7 @@ const (
 
 var (
 	// Use buckets ranging from 5 ms to 2.5 seconds (admission webhooks timeout at 30 seconds by default).
-	latencyBuckets       = []float64{0.005, 0.025, 0.1, 0.5, 2.5}
+	latencyBuckets       = []float64{0.005, 0.025, 0.05, 0.1, 0.5, 1.0, 2.5, 5.0, 10.0, 15.0, 20.0, 25.0, 30.0}
```

Comment:
I believe that the top bucket is set to 2.5s because (as confirmed with @liggitt), admission webhooks have a default timeout set to 10s. Practically, this means that we have discrete buckets up to 2.5s and the webhooks which take longer than 2.5s (which is already pretty slow) will fall into the remaining infinite bucket. 

Considering the fact that the apiserver already has a default request timeout set to 60s (again, confirmed with @liggitt), I'm not sure how much value this change is going to bring.

Reply:
I'm okay with adding smaller buckets, for better granularity but bigger buckets probably don't really help much. Personally, I'd flag any webhook request taking over 2.5s as problematic.

Reply:
Yes - webhooks shouldn't be taking more than at most small hundreds of milliseconds (even that is a lot). So I agree with the above.
[Adding higher granularity on lower values makes send though.]

Reply:
Hi @logicalhan @wojtek-t, thanks for your feedback. I totally agree that webhooks should be really fast and should not take more than a few hundreds of milliseconds.

I think having buckets covering up to the timeout (in this case `10s` for admission webhooks) provides a little bit more accurate picture than a `2.5s` flat line we are seeing right now:
- With a `2.5s` flat line we don't know whether the admission webhook is being timed out by the apiserver. It could be that it's taking more than `2.5s` but it is well within the timeout boundary.
- If we put `apiserver_request_duration_seconds`, `etcd_request_duration_seconds` and `admission_duration_seconds` side by side then we have a little bit more accurate picture of latency breakdown (if the buckets for all these metrics cover the timeout).

I think, if we define the timeout to be `10s` then we also want to measure up to `10s`. I propose the following - higher granularity on lower values and much lower granularity on the higher values leading up to the timeout : `[]float64{0.005, 0.01, 0.025, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5, 1.0, 2.5, 5.0, 10.0}`

Please let me know if this looks good.
Thanks!

Reply:
I don't have strong opinion.
@logicalhan - WDYT?

Reply:
Ah, sorry, this fell off my plate. Yeah that's fine with me. I would omit the 10s bucket though, since 5s is going to already be pretty problematic (Mutating webhooks are run twice, so with a 5s avg latency, it'll hit the timeout).